api/src/main/java/org/openmrs/util/ConceptReferenceRangeUtility.java

```diff
@@ -0,0 +1,248 @@
+			criteria = criteria.replace("$fn.getAgeInDays()", "$fn.getAgeInDays($patient)");
+		}
+		return criteria;
+	}
+	
+	/**
+	 * Method to get the age of a person in months.
+	 *
+	 * @param person patient in subject
+	 * @return the age in months as an Integer
+	 * 
+	 * @since 2.7.0
+	 */
+	public Integer getAgeInMonths(Person person) {
+		return getAge(ChronoUnit.MONTHS, person);
+	}
+	
+	/**
+	 * Method to get the age of a person in weeks.
+	 *
+	 * @param person patient in subject
+	 * @return the age in weeks as an Integer
+	 *
+	 * @since 2.7.0
+	 */
+	public Integer getAgeInWeeks(Person person) {
+		return getAge(ChronoUnit.WEEKS, person);
+	}
+	
+	/**
+	 * Method to get the age of a person in days.
+	 *
+	 * @param person patient in subject
+	 * @return the age in days as an Integer
+	 * 
+	 * @since 2.7.0
+	 */
+	public Integer getAgeInDays(Person person) {
+		return getAge(ChronoUnit.DAYS, person);
+	}
+	
+	/**
+	 * Method to get the age of a person in years.
+	 *
+	 * @param person patient in subject
+	 * @return the age in years as an Integer
+	 * 
+	 * @since 2.7.0
+	 */
+	public Integer getAge(Person person) {
+		return getAge(ChronoUnit.YEARS, person);
+	}
+	
+	/**
+	 * Gets the age of a person with a specified ChronoUnit.
+	 *
+	 * @param chronoUnit the unit of precision for the age calculation (e.g. WEEKS, MONTHS, YEARS)
+	 * @param person the person/patient to get age from   
+	 * @return the age in the specified unit as an Integer
+	 * 
+	 * @since 2.7.0
+	 */
+	public Integer getAge(ChronoUnit chronoUnit, Person person) {
+		if (person == null || person.getBirthdate() == null) {
+			return null;
+		}
+		
+		LocalDate birthDate = new java.sql.Date(person.getBirthdate().getTime()).toLocalDate();
+		LocalDate endDate = LocalDate.now();
+		
+		// If date given is after date of death then use date of death as end date
+		if (person.getDeathDate() != null) {
+			LocalDate deathDate = new java.sql.Date(person.getDeathDate().getTime()).toLocalDate();
+			if (endDate.isAfter(deathDate)) {
+				endDate = deathDate;
+			}
+		}
+		
+		long age;
+		
+		switch (chronoUnit) {
+			case DAYS:
+				age = ChronoUnit.DAYS.between(birthDate, endDate);
+			case WEEKS:
+				age = ChronoUnit.WEEKS.between(birthDate, endDate);
+				break;
+			case MONTHS:
+				age = ChronoUnit.MONTHS.between(birthDate, endDate);
+				break;
+			case YEARS:
+				age = ChronoUnit.YEARS.between(birthDate, endDate);
+				break;
+			default:
+				throw new IllegalArgumentException("Unsupported ChronoUnit: " + chronoUnit);
+		}
+		
+		return (int) age;
+	}
+	
+	/**
+	 * Gets the latest Obs by concept.
+	 *
+	 * @param conceptKey can be either concept uuid or conceptMap's code and sourceName 
+	 *                   e.g "bac25fd5-c143-4e43-bffe-4eb1e7efb6ce" or "CIEL:1434"
+	 *                   
+	 * @return Obs
+	 * 
+	 * @since 2.7.0
+	 */
+	public Obs getLatestObsByConcept(String conceptKey) {
+		Concept concept;
+
+		if (conceptKey.contains(":")) {
```

Comment:
Why not just use `Context.getConceptService().getConceptByReference()`?

Reply:
Good to know that this method exists!